images/macos/software-report/SoftwareReport.Generator.ps1

```diff
@@ -22,311 +23,246 @@ Import-Module "$PSScriptRoot/../helpers/Xcode.Helpers.psm1"
-    (Get-VcpkgVersion)
-)
-
-$markdown += New-MDList -Style Unordered -Lines ($packageManagementList | Sort-Object)
-$markdown += New-MDHeader "Environment variables" -Level 4
-$markdown += Build-PackageManagementEnvironmentTable | New-MDTable
-$markdown += New-MDNewLine
+$packageManagement = $installedSoftware.AddHeaderNode("Package Management")
+$packageManagement.AddToolNode("Bundler", $(Get-BundlerVersion))
+$packageManagement.AddToolNode("Carthage", $(Get-CarthageVersion))
+$packageManagement.AddToolNode("CocoaPods", $(Get-CocoaPodsVersion))
+$packageManagement.AddToolNode("Composer", $(Get-ComposerVersion))
+$packageManagement.AddToolNode("Homebrew", $(Get-HomebrewVersion))
+$packageManagement.AddToolNode("Miniconda", $(Get-CondaVersion))
+$packageManagement.AddToolNode("NPM", $(Get-NPMVersion))
+$packageManagement.AddToolNode("NuGet", $(Get-NuGetVersion))
+$packageManagement.AddToolNode("Pip 2", $(Get-PipVersion -Version 2))
+$packageManagement.AddToolNode("Pip 3", $(Get-PipVersion -Version 3))
+$packageManagement.AddToolNode("Pipx", $(Get-PipxVersion))
+$packageManagement.AddToolNode("RubyGems", $(Get-RubyGemsVersion))
+$packageManagement.AddToolNode("Vcpkg", $(Get-VcpkgVersion))
+$packageManagement.AddToolNode("Yarn", $(Get-YarnVersion))
+
+$packageManagement.AddNode($(Build-PackageManagementEnvironmentTable))
 
 # Project Management
-$markdown += New-MDHeader "Project Management" -Level 3
-$markdown += New-MDList -Style Unordered -Lines (@(
-    (Get-MavenVersion),
-    (Get-GradleVersion),
-    (Get-ApacheAntVersion),
-    (Get-SbtVersion)
-    ) | Sort-Object
-)
+$projectanagement = $installedSoftware.AddHeaderNode("Project Management")
+$projectanagement.AddToolNode("Apache Ant", $(Get-ApacheAntVersion))
+$projectanagement.AddToolNode("Apache Maven", $(Get-MavenVersion))
+$projectanagement.AddToolNode("Gradle", $(Get-GradleVersion))
+$projectanagement.AddToolNode("Sbt", $(Get-SbtVersion))
 
 # Utilities
-$markdown += New-MDHeader "Utilities" -Level 3
-$utilitiesList = @(
-    (Get-CurlVersion),
-    (Get-GitVersion),
-    (Get-GitLFSVersion),
-    (Get-GitHubCLIVersion),
-    (Get-HubVersion),
-    (Get-WgetVersion),
-    (Get-SVNVersion),
-    (Get-PackerVersion),
-    (Get-OpenSSLVersion),
-    (Get-JqVersion),
-    (Get-PostgresClientVersion),
-    (Get-PostgresServerVersion),
-    (Get-Aria2Version),
-    (Get-AzcopyVersion),
-    (Get-ZstdVersion),
-    (Get-BazelVersion),
-    (Get-BazeliskVersion),
-    (Get-MongoVersion),
-    (Get-MongodVersion),
-    (Get-7zipVersion),
-    (Get-BsdtarVersion),
-    (Get-GnuTarVersion),
-    (Get-GPGVersion),
-    (Get-SwitchAudioOsxVersion),
-    (Get-SoxVersion),
-    (Get-YqVersion),
-    (Get-ImageMagickVersion)
-)
-
-if ($os.IsLessThanMonterey) {
-    $utilitiesList += @(
-        (Get-HelmVersion)
-    )
+$utilities = $installedSoftware.AddHeaderNode("Utilities")
+$utilities.AddToolNode("7-Zip", $(Get-7zipVersion))
+$utilities.AddToolNode("aria2", $(Get-Aria2Version))

```

Comment:
If we return `Node` from every method instead of version string, we can simplify this file a bit. It will look something like this:
```pwsh
$utilities = $installedSoftware.AddHeaderNode("Utilities")
$utilities.AddNodes(@(
  $(Get-AzcopyVersion),
  $(Get-BazelVersion),
  $(Get-BazeliskVersion),
  $(Get-BsdtarVersion),
  $(Get-CurlVersion),
  $(Get-GitVersion),
  $(Get-GitLFSVersion),
  $(Get-GitHubCLIVersion)
))
```

Don't push to rework the current approach. Just thinking what approach will work better for us to maintain.

Reply:
This may look very nice, but let discuss it a bit later